compiler/parser/src/parser/expression.rs

```diff
@@ -252,11 +252,33 @@ impl ParserContext<'_> {
         }
         let mut inner = self.parse_postfix_expression()?;
         for (op, op_span) in ops.into_iter().rev() {
-            inner = Expression::Unary(UnaryExpression {
-                span: op_span + inner.span(),
-                op,
-                receiver: Box::new(inner),
-            });
+            inner = match inner {
+                // If the unary operation is a negate, and the inner expression is a signed integer literal,
+                // then produce a negative integer literal.
+                // This helps handle a special case where -128i8, treated as a unary expression, overflows, but -128i8, treated as an integer literal doesn't.
+                // TODO: the parser produces a negative integer literal for both -(128i8) and -128i8. Is this an issue?
```

Comment:
I would advocate for only allowing `-128i8` to denote the negative literal. This is what the Leo Reference says, at least. We could change the Leo Reference, but then we would need to allow any amount of parenthesization `-((…(128i8)…))`.

Reply:
I wouldn’t go as far as requiring no whitespace or comments between the `-` and the `128i8`, because that would complicate the lexical story for no good reason.

Reply:
Agreed, the new implementation only performs the transformation for `-`, `Token::Integer(..)`